### ISSUE TITLE
Handle circular references

### ISSUE THREAD

**Issue by EugenCepoi** [owlike/genson#59]
One simple way would be to make use of the metadata system to write at the begining of the json object an @id and at the places where the reference happens use @ref that looks back.
The downside is that on the other end it might be tricky to deser if someone uses another lib...

Another alternative would be to use a json pointer expression...

This might have a performance impact, we should provide a way to enable it explicitly via annotations and globally with the builder. When enabled globally we can optimize it further by enabling it only for cycles in the types (so potential object cycle) and untyped objects.

**Comment by OrionDevelopment** [comment]
Is this in process?

**Comment by EugenCepoi** [comment]
No, but contributions would be welcome. I can find some time to help but I don't have enough time to do it my self.
